### ISSUE TITLE
Errors while importing Email template in Node

### ISSUE THREAD

**Issue by moizkamran** [resend/react-email#812]
## Bug Description

I am facing an issue while integrating the email functionality using `@react-email/render` and `nodemailer` packages. The code crashes with an "ERR_MODULE_NOT_FOUND" error.

## Actual Behavior

The code crashes with an "ERR_MODULE_NOT_FOUND" error. The error message indicates that a specific module is not found. Error: module not found
    at ...
    ...

## Environment

- Node.js version:
- `@react-email/render` version:
- `nodemailer` version:

## Additional Information

- Any modifications or customizations made to the email template or code.
- Any relevant configuration settings or environment variables used.
- Any other details or observations that could be helpful.

```
// Code snippet or relevant configurations
import User from '../schemas/User.schema.js';
import { Email } from '../emails/email';
import { render } from '@react-email/render';


const userRoute = async (fastify) => {

    const giftHTML = render(Email);

    fastify.post('/', async (request, reply) => {
        try {
            const userData = request.body
            
            // Create a new user object based on the received data
            const user = new User(userData)
```

### Link to the code that reproduces this issue

https://react.email/docs/integrations/nodemailer#3-convert-to-html-and-send-email

## Steps to Reproduce

1. Install dependencies: `@react-email/render` and `nodemailer`.
2. Create an email template using React in a `.jsx` or `.tsx` file.
3. Import the email template, convert it to HTML, and send the email using Nodemailer.
4. Execute the code and observe the crash with the "ERR_MODULE_NOT_FOUND" error.

## Expected Behavior

This is what react-email says on their docs to convert emails into html with node js
<img width="744" alt="image" src="https://github.com/resendlabs/react-email/assets/70318057/694cead3-e476-47fe-a1fe-c1553d9f1b79">
``

**Comment by liydaco** [comment]
Same issue, node v 18.14.0

**Comment by joaopcm** [comment]
Hey! I'm facing the same issue! Is there any update on this?

**Comment by gabrielmfern** [comment]
Hey @joaopcm, since you were the last one to comment here, could you make a minimal reproduction repo of this?

**Comment by joaopcm** [comment]
> Hey @joaopcm, since you were the last one to comment here, could you make a minimal reproduction repo of this?

Hey, @gabrielmfern! Thank you for reminding me I left this comment in this thread! I found the solution and shared it [here](https://twitter.com/jopcmelo/status/1771312930639741085). My issue was more about supporting React.js components in a Node.js environment using a NestJS project.

**Comment by Nithishprem** [comment]
Hey, I am also difficulty in using react-email components in nodejs server.
Is it possible to import react-email components in **express** **sever** to send emails?

**Comment by joaopcm** [comment]
> Hey, I am also difficulty in using react-email components in nodejs server. Is it possible to import react-email components in **express** **sever** to send emails?

Hey, @Nithishprem! I believe I had a similar issue. I was using NestJS (which uses Express as the main API engine) with TypeScript. As I'm using SWC to build my project, I needed to add the proper configurations to support JSX/TSX files when compiling the project. Have you already tried something similar to [this](https://twitter.com/jopcmelo/status/1771312930639741085)?

**Comment by shon-morgun** [comment]
same issue node v18.17.0

**Comment by LockeandDropit** [comment]
Hey @joaopcm!

Both links you provided redirect to a twitter (X) post that doesn't have any comments or solutions present. Were you able to find a solution to the above issue? I have been stuck on it for longer than I care to admit 😅

**Comment by joaopcm** [comment]
> Hey @joaopcm!
> 
> Both links you provided redirect to a twitter (X) post that doesn't have any comments or solutions present. Were you able to find a solution to the above issue? I have been stuck on it for longer than I care to admit 😅

@LockeandDropit, this is what I see when I open the X link ↓

![CleanShot 2024-05-28 at 4  06 01@2x](https://github.com/resend/react-email/assets/58827242/634f0d9d-d142-47f0-98a0-4d5cf76e2807)

**Comment by gabrielmfern** [comment]
@gusmagnago Could you make a reprodution of it? Seems to me that everyone having this issue here is having it for different reasons.

It could be that you might have something misconfigured, as @joaopcm did.

**Comment by basememara** [comment]
We're having issues too getting our NestJS backend to use React Email to send out emails.

**Comment by gabrielmfern** [comment]
@basememara Would you be willing to share a minimal reproduction of it? With one I can help you for sure.

**Comment by gabrielmfern** [comment]
Closing this as stale, if anyone is willing to share a reproduction of it, I'll be more than happy to reopen and fix.

**Comment by danLeBrown** [comment]
I ran into the same issue today, but I was able to set it up eventually and send emails successfully. My setup is something like this:

_tsconfig.json_
``json {
"compilerOptions": {
...,
    "jsx": "react-jsx",
    "allowUmdGlobalAccess": true
},
  "include": ["src/*", "src/**/*", "src/domains/**/*"],
  "exclude": ["node_modules", "**/*.spec.ts", "dist", "build"]
}
```

_package.json_
```json 
{
...,
  "dependencies": {
...,
    "@react-email/components": "0.0.31",
    "@react-email/render": "1.0.3",
    "react": "19.0.0",
    "react-dom": "19.0.0",
  },
  "devDependencies": {
...,
    "@types/react": "19.0.1",
    "@types/react-dom": "19.0.1",
    "react-email": "3.0.4",
  }
}
```

_service where I import the template_

```ts
import { Injectable } from '@nestjs/common';
import { render } from '@react-email/render';
import StripeWelcomeEmail from '../templates/stripe-welcome'; // notice there's no .tsx extension here
import { MailTrap } from './providers/mailtrap.service';

@Injectable()
export class MailsService {
  constructor(private provider: MailTrap) {}

  async sendWelcomeEmail(to: string) {
    const html = await render(StripeWelcomeEmail());

    await this.provider.sendEmail({
      from: 'no-reply@example.com',
      to,
      subject: 'Welcome to Example!',
      html,
    });
  }
}
```

I hope this comes in handy for anyone interested

**Comment by gabrielmfern** [comment]
@danLeBrown what was the root cause in your case?

**Comment by danLeBrown** [comment]
@gabrielmfern, my issues were primarily resolved by updating the `tsconfig` compiler options, which were mainly related to Node.js’s default inability to import `tsx` files. Additionally, I encountered challenges with some React-related libraries, if I recall correctly.

Later on, I faced an npm dependency issue: a particular requirement was taking too long to install and kept timing out. I resolved this by manually running `pnpm i -D @next/swc-linux-x64-gnu --save-optional` rather than relying on the dependency’s automated installation.
